.github/workflows/awesome_workflow.yml

```diff
@@ -15,8 +15,8 @@ jobs:
       - uses: actions/setup-python@v2
       - name: requirements
         run: | 
-          sudo apt -qq -y update
-          sudo apt -qq install clang-tidy-10 clang-format-10
+          sudo apt-get -qq update
+          sudo apt-get -qq install clang-tidy clang-format
```

Comment:
For some reason I can't add suggestion to the whole file, but this looks good except the use of your personal credentials. I would rather you use the `permissions` block at the top level scope (where `jobs` is defined).

For example:

```yml
name: Awesome CI Workflow
on: [push, pull_request]
permissions:
  contents: write # this should suffice I think
jobs:
  MainSequence:
    name: Code Formatter
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1 # v2 is broken for git diff
       - uses: actions/setup-python@v2
       - name: requirements
         run: | 
           sudo apt-get -qq update
           sudo apt-get -qq install clang-tidy clang-format
        # checks are passing with less errors when used with this version. 
         # The default installs v6.0 which did not work out well in my tests
       - name: Setup Git Specs
        run: |
          git config --global user.name github-actions[bot] # this and the email should be different from an actual user
          git config --global user.email 'github-actions@users.noreply.github.com'
          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
     ...
```

Reply:
I didn't know that was possible 🤔 this is the reason I was using my credentials to make it easier 😅 A few weeks back I read about this and said this wasn't possible. Let's give it a try, though.